plugins/modules/nd_api_key.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Dev Sinha (@DevSinha13) <devsinh@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+from copy import deepcopy
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+
+DOCUMENTATION = r"""
+---
+module: nd_api_key
+version_added: "1.4.0"
+short_description: Manage API keys in Nexus Dashboard
+description:
+- This module allows you to manage API keys in Cisco Nexus Dashboard.
+- It supports creating, updating, querying, and deleting API keys.
+- API key names must be 1-32 characters long and can only contain letters, digits, '_', '.', and '-'.
+author:
+- Dev Sinha (@DevSinha13)
+options:
+  api_key_id:
+    description:
+    - The ID of the API key to manage.
+    - Required when state is 'absent' if api_key_name is not specified.
+    type: str
+  api_key_name:
+    description:
+    - The name/description of the API key.
+    - Required when state is 'present'.
+    - Must be 1-32 characters long.
+    - Only alphanumeric characters, underscores, periods, and hyphens are allowed.
+    type: str
+    aliases: [ description ]
+  annotations:
+    description:
+    - Key-value pairs of annotations to add to the API key.
```

Comment:
Do we allow any value type in the k/v pair? If only strings should we restrict the input by providing a list of dict with the key and value as string options?

```suggestion
    - The annotations attached to the API key.
    - ... explanation of what it is ...
    - ... perhaps an example...
```